src/others/helpers/map-utils.ts

```diff
@@ -1,160 +1,51 @@
-        locationAndDateRequests.aidRequests.forEach((req) =>
-          addFeatureForAidRequest(req, locationGroup.location, locationAndDateRequests.date, features)
-        );
-        addAggregatedFeaturePerLocationAndDate(locationGroup, locationAndDateRequests, features);
-      });
-    });
-  });
-  const maxPerCategory: { [id: string]: number } = {};
-  features.forEach(function (feature) {
-    if (feature.properties) {
-      if (!(feature.properties.category in maxPerCategory)) {
-        maxPerCategory[feature.properties.category] = 0;
-      }
-      maxPerCategory[feature.properties.category] = Math.max(maxPerCategory[feature.properties.category], feature.properties.amount);
-    }
-  });
-  const featuresWithNormalizedAmount = Array.from(features).map((feature) => {
-    if (feature.properties) {
-      feature.properties.normalized_amount = feature.properties.amount / maxPerCategory[feature.properties.category];
-    }
-    return feature;
-  });
-  return featuresWithNormalizedAmount;
-};
-
-export const possibleDates = (requests: AidRequest[]) => {
-  return requests.map((req) => req.date).filter((value, index, self) => self.indexOf(value) === index);
-};
-
-function addAggregatedFeaturePerLocationAndDate(
-  locationGroup: DecodedAidRequestGroupedByLocation,
-  dateGroup: GroupedByDateWithTotal,
-  features: Set<Feature<Geometry, RequestProperties>>
-) {
-  features.add({
-    type: "Feature",
-    properties: {
-      amount: dateGroup.total,
-      date: dateGroup.date,
-      category: "ALL",
-      description: dateGroup.totalDescription,
-      city: locationGroup.location.name,
-    },
-    geometry: { type: "Point", coordinates: [locationGroup.location.lon, locationGroup.location.lat] },
-  });
-}
-
-function addFeatureForAidRequest(
-  aidRequest: Omit<DecodedAidRequest, "date">, // TODO: consider replacing with AidRequest
-  location: Location,
-  date: string,
-  features: Set<Feature<Geometry, GeoJsonProperties>>
-) {
-  features.add({
-    type: "Feature",
-    properties: {
-      amount: aidRequest.amount,
-      date: date,
-      category: aidRequest.name,
-      description: aidRequest.name + ": " + aidRequest.amount,
-      city: location.name,
-    },
-    geometry: { type: "Point", coordinates: [location.lon, location.lat] },
+export const mapAidRequestsToFeatures = (mapData: MapData[], locationTranslator: (city_id: number) => Location | undefined): Feature<Geometry, GeoJsonProperties>[] => {
+  const maxAmount = mapData.reduce((max, dataPoint) => Math.max(max, dataPoint.amount), 0);
+  return mapData.map((dataPoint) => {
+    const location = locationTranslator(dataPoint.city_id);
+    if (!location) throw new Error(`Location: ${dataPoint.city_id} is not found`);
+
+    return {
+        type: "Feature",
+        properties: {
+          amount: dataPoint.amount,
+	  city: location.name,
+	  normalized_amount: dataPoint.amount / maxAmount,
```

Comment:
**question**

What do wee need `normalized_amount` for?

Reply:
To compute colors of the regions.

Reply:
Then it shouldn't be here, but in the module reponsible for mapping mapdata to layer styles.

Reply:
This is the function that creates data that is fed to the map. So this computation needs to be here.

Reply:
Or at the very least, in a separately declared function, with a specified API and a test or two. 

So you could use dependency injection and use it later wherever you want.

Reply:
This function has a test and it has a specified API (modulo the fact that you have a complete freedom of putting any metadata you with in the 'properties').

I don't really get why doing "separately declared function" / "dependency injection" for a one line change:
"normalized_amount: dataPoint.amount / maxAmount".

What is so special about this field? Why are you not complaining about amount / city / description fields?

Reply:
Offline conclusion: Maciej will send a separate PR with his proposed changes.